src/server.cc

```diff
@@ -1422,3 +1412,24 @@ void Server::AdjustOpenFilesLimit() {
                  << " (it's originally set to " << old_limit << ")";
   }
 }
+
+std::string ServerLogData::Encode() {
+  if (type_ == kReplIdLog) {
+    return std::string(1, kReplIdTag) + " " + content_;
+  } else {
+    return content_;
+  }
+}
+
+Status ServerLogData::Decode(const rocksdb::Slice &blob) {
+  if (blob.size() == 0) {
+    return Status(Status::NotOK);
+  }
+
+  const char *header = blob.data();
+  if (*header == kReplIdTag) {
+    type_ = kReplIdLog;
+  }
+  content_ = std::string(blob.data()+2, blob.size()-2);
```

Comment:
should check the blob data length before using it?

Reply:
ok, it is more strict